aten/src/ATen/native/Convolution.cpp

```diff
@@ -443,8 +468,14 @@ at::Tensor _convolution(
     AT_CHECK(!bias.defined() || (input.type() == bias.type()),
              "Input type (", input.type().toString(), ") and bias type (", bias.type().toString(),
              ") should be the same");
-    output = at::mkldnn_convolution(input, weight.contiguous(), bias.defined() ? bias.contiguous() : bias,
-                                    params.padding, params.stride, params.dilation, params.groups);
+    if (!input_is_mkldnn) {
+      output = at::mkldnn_convolution(input, weight.contiguous(), bias.defined() ? bias.contiguous() : bias,
+                                      params.padding, params.stride, params.dilation, params.groups);
+    } else {
+      // do not call contiguous on mkldnn tensor
+      output = at::mkldnn_convolution(input, weight, bias,
+                                      params.padding, params.stride, params.dilation, params.groups);
+    }
 #endif
```

Comment:
also once we have `contiguous()`, we don't have to do if-else here depending on `input_is_mkldnn`, just one call to `at::mkldnn_convolution` is needed